src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -1229,23 +1233,126 @@ private int SkipBlockComment(int i)
+                case 'r': return "\r";
+                case 't': return "\t";
+                case 'u': return ScanUnicodeEscapeSequence();
+                case 'v': return "\v";
+                default: return c.ToString();
+            }
+        }
+
+        private string ScanUnicodeEscapeSequence()
+        {
+            int escSeqStartIndex = _currentIndex - 2;
+            int maxNumberOfHexDigits = 6;
+            bool isTerminated = false;
+
+            char c = GetChar();
+            if (c != '{')
+            {
+                UngetChar();
+
+                IScriptExtent errorExtent = NewScriptExtent(escSeqStartIndex, _currentIndex);
+                ReportError(errorExtent, () => ParserStrings.InvalidUnicodeEscapeSequence);
+                return s_unknownUnicodeChar;
+            }
+
+            // Scan hex chars after the Unicode escape sequence start.
+            var sb = GetStringBuilder();
+            int i;
+            for (i = 0; i < maxNumberOfHexDigits; i++)
+            {
+                c = GetChar();
+
+                // Sequence has been terminated.
+                if (c == '}')
+                {
+                    if (i == 0)
+                    {
+                        // Sequence must have at least one hex char.
+                        IScriptExtent errorExtent = NewScriptExtent(escSeqStartIndex, _currentIndex);
+                        ReportError(errorExtent, () => ParserStrings.InvalidUnicodeEscapeSequence);
+                        return s_unknownUnicodeChar;
+                    }
+
+                    isTerminated = true;
+                    break;
+                }
+
+                if (!c.IsHexDigit())
+                {
+                    UngetChar();
+
+                    ReportError(_currentIndex, () => ParserStrings.MissingUnicodeEscapeSequenceTerminator);
+                    return s_unknownUnicodeChar;
+                }
+
+                sb.Append(c);
+            }
+
+            if (!isTerminated)
+            {
+                c = GetChar();
+                if (c != '}')
+                {
+                    UngetChar();
+                    ReportError(_currentIndex,
+                        i == maxNumberOfHexDigits && c.IsHexDigit()
+                            ? (Expression<Func<string>>)(() => ParserStrings.TooManyDigitsInUnicodeEscapeSequence)
+                            : () => ParserStrings.MissingUnicodeEscapeSequenceTerminator);
+                    return s_unknownUnicodeChar;
+                }
+            }
+
+            string hexStr = GetStringAndRelease(sb);
+
+            int unicodeValue = int.Parse(hexStr, NumberStyles.AllowHexSpecifier, NumberFormatInfo.InvariantInfo);
+            if (unicodeValue <= Char.MaxValue)
+            {
+                return ((char)unicodeValue).ToString();
+            }
+            else if (unicodeValue <= 0x10FFFF)
+            {
+                return Char.ConvertFromUtf32(unicodeValue);
```

Comment:
We've removed the fixed width format completely.  What I was demonstrating above was a surrogate pair (in the 0xD800-0xDFFF reserved range) using `C#` syntax.  And yes, you can use `` `u{1F44D}``.  You can even use `` `u{01F44D}`` if you'd like.  We stop at 6 hex digits for now because the max value you can use for Unicode - even with surrogate pairs is 0x10FFFF which requires no more than six hex digits to encode.  Any more hex digits would have to be 0 and therefore  a bit pointless at this time.  And one advantage of the brace syntax is that it could be extended in the future to include more digits **if** the Unicode range increased.  As for excluding direct use of surrogate characters, I see no good reason to do that.